test/timecop_test.rb

```diff
@@ -539,6 +539,21 @@ def test_date_strptime_without_specifying_format
     end
   end
 
+  def test_date_strptime_with_day_of_week
+    Timecop.freeze(Time.new(1984,2,28)) do
+      assert_equal Date.strptime('Thursday', '%A'), Date.new(1984, 3, 1)
+      assert_equal Date.strptime('Monday', '%A'), Date.new(1984, 2, 27)
+    end
+  end
+
+  def test_date_strptime_with_invalid_date
+    begin
+      Date.strptime('', '%Y-%m-%d')
+    rescue ArgumentError => e
```

Comment:
Better to use `assert_raises`: https://apidock.com/ruby/Test/Unit/Assertions/assert_raise